subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -47,8 +44,7 @@
     private final Condition workDone = lock.newCondition();
     private State state = State.Working;
     private int workers;
-    private final Set<OperationHolder> notFinished = new HashSet<OperationHolder>();
-    private final LinkedList<OperationHolder> notYetStarted = new LinkedList<OperationHolder>();
+    private final LinkedList<T> workQueue = new LinkedList<T>();
```

Comment:
Just a minor thing: you could use a `Deque` here, which would make the `poll` / `pick` from queue clearer.